### PR TITLE
Remove cuda cub

### PR DESCRIPTION
**Pull request by griwodz**
This PR is meant to replace #103 . That PR attempts to fix a thrust::sort linker problem for sm5X that is due to an old bug in Thrust. 

This is a simpler solution: CUB is removed entirely, the Thrust linker bug is fixed in an unexpected way: instead of fixing thrust::sort, just provide a StrictWeakOrdering operator implemented in a way that guarantees that it doesn't use any memory.

### CODE REVIEW

**Comment by griwodz** [comment]
AppVeyor could no longer find packages. I followed recommendations for upgrading AppVeyor and hope that I didn't break anything badly.

File: src/cctag/cuda/frame_07d_vote_if.cu [hunk 1]
```diff
@@ -64,140 +59,35 @@ bool Frame::applyVoteIf( )
 
     NumVotersIsGreaterEqual select_op( _voters.dev );
 
+#if 0
```
Comment (simogasp) [inline src/cctag/cuda/frame_07d_vote_if.cu]: maybe we could define a variable to activate this piece of code from cmake and report this bug in a `known issues` section in some of the doc file (README or BUILD)?
```cmake
option(CCTAG_NO_THRUST_COPY_IF "Do not use thrust::copy_if(), circumvent bugs in cuda 7 +  GTX 980 and GTX 980 Ti GTX 1080  " OFF)

# then is the if(CCTAG_WITH_CUDA) part
if(CCTAG_NO_THRUST_COPY_IF)
target_compile_definitions(CCTag  PRIVATE -DCCTAG_NO_THRUST_COPY_IF)
endif()
                         
# also add the entry in the config.hpp.in
```

Comment (griwodz) [reply src/cctag/cuda/frame_07d_vote_if.cu]: Good idea, just pushed.

**Comment by simogasp** [comment]
LGTM, if someone manages to build it on a windoze system just to be sure we can merge.

**Comment by griwodz** [comment]
> LGTM, if someone manages to build it on a windoze system just to be sure we can merge.

Well, it was "fixed" by asking AppVeyor for the "Visual Studio 2017" image instead of the "Visual Studio 2015" image. Other people have used "Previous Visual Studio 2015" as image. I hope that's good enough for now?

**Pull request merged**